### ISSUE TITLE
ECS Metrics not working on Fargate

### ISSUE THREAD

**Issue by wyTrivail** [aws-observability/aws-otel-collector#127]
ECS Metrics test was failing on FARGATE.

the platform version is 1.4.0.

this is the testing result.

```
module.validator_without_sample_app[0].null_resource.validator (local-exec): validator_1  | com.amazon.aoc.exception.BaseException: metric in toBeCheckedMetricList {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: aoc-testing-e51a5848085cf11c}, {Name: ecs.service,Value: undefined}, {Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}, {Name: ecs.task-definition-version,Value: 1}]} not found in baseMetricList: [{Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.reserved_e51a5848085cf11c,Dimensions: []}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}, {Name: ecs.service,Value: undefined}, {Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}, {Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.service,Value: undefined}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.utilized_e51a5848085cf11c,Dimensions: []}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}, {Name: ecs.service,Value: undefined}, {Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}, {Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.service,Value: undefined}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.cpu.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.reserved_e51a5848085cf11c,Dimensions: []}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}, {Name: ecs.service,Value: undefined}, {Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}, {Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.service,Value: undefined}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.reserved_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.utilized_e51a5848085cf11c,Dimensions: []}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}, {Name: ecs.service,Value: undefined}, {Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}, {Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.service,Value: undefined}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.memory.utilized_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.rx_e51a5848085cf11c,Dimensions: []}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.rx_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}, {Name: ecs.service,Value: undefined}, {Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}, {Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.rx_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.rx_e51a5848085cf11c,Dimensions: [{Name: ecs.service,Value: undefined}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.rx_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.rx_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.tx_e51a5848085cf11c,Dimensions: []}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.tx_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}, {Name: ecs.service,Value: undefined}, {Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}, {Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.tx_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.tx_e51a5848085cf11c,Dimensions: [{Name: ecs.service,Value: undefined}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.tx_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.network.rate.tx_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.read_bytes_e51a5848085cf11c,Dimensions: []}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.read_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}, {Name: ecs.service,Value: undefined}, {Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}, {Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.read_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.read_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.service,Value: undefined}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.read_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.read_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.write_bytes_e51a5848085cf11c,Dimensions: []}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.write_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}, {Name: ecs.service,Value: undefined}, {Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}, {Name: ecs.task-definition-version,Value: 1}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.write_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.cluster,Value: arn:aws:ecs:us-west-2:***:cluster/aoc-testing-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.write_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.service,Value: undefined}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.write_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-family,Value: taskdef-e51a5848085cf11c}]}, {Namespace: aws-otel/aws-otel-integ-test,MetricName: ecs.task.storage.write_bytes_e51a5848085cf11c,Dimensions: [{Name: ecs.task-definition-version,Value: 1}]}]
```

**Comment by hossain-rayhan** [comment]
Hi @wyTrivail , I tested the image with Fargate locally and found all the metrics in CloudWatch. However, it would be easy to debug if can answer the followings for me.
1. Where did you find this?
2. How to reproduce it?
3. Wondering, why didn't we catch before merging it? Can we do something so that we can find this earlier. Maybe I missed something.

**Comment by wyTrivail** [comment]
Talked with Rayhan offline and found the root cause, which is ecsmetric receiver doesn't properly handle the cluster name getting from Fargate. will wait for Rayhan to fix it before release collector.

To answer the question 3,  we can't run the integration test on the PR before merge as it would leak our aws credentials. And you can definitely do something to catch it before merging as you have done using testing framework locally on the EC2 type of ECS, however looks like you didn't run Fargate to validate. 

Additionally, i wish we can add some unit tests for this kind of errors when you submit the PR to fix it.

**Comment by hossain-rayhan** [comment]
PR to address this: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1626

**Comment by hossain-rayhan** [comment]
Closed by #137
